src/dbnode/persist/fs/retriever.go

```diff
@@ -372,18 +367,43 @@ func (r *blockRetriever) fetchBatch(
 	blockStart time.Time,
 	allReqs []*retrieveRequest,
 	seekerResources ReusableSeekerResources,
+	retrieverResources *reuseableRetrieverResources,
 ) {
-	// Resolve the seeker from the seeker mgr
-	seeker, err := seekerMgr.Borrow(shard, blockStart)
-	if err != nil {
+	var (
+		seeker     ConcurrentDataFileSetSeeker
+		callbackWg sync.WaitGroup
+	)
+
+	defer func() {
+		filteredReqs := allReqs[:0]
```

Comment:
Nit: after the last changes, it would make sense to rename `filteredReqs` to `pendingReqs` or something like that.

Reply:
:+1: will roll this up into the followup